src/pymor/reductors/neural_network.py

```diff
@@ -466,17 +546,53 @@ def compute_training_data(self):
             for datum in self.training_set:
                 if not self.fom:
                     mu, output = datum
-                    sample = self._compute_sample(mu, output=output)
+                    samples = self._compute_sample(mu, output=output)
                 else:
-                    sample = self._compute_sample(datum)
-                self._update_scaling_parameters(sample)
-                self.training_data.extend(sample)
+                    samples = self._compute_sample(datum)
+                for sample in samples:
+                    self._update_scaling_parameters(sample)
+                self.training_data.extend(samples)
+
+    def compute_validation_data(self):
+        """Compute the training samples (the outputs to the parameters of the training set)."""
+        with self.logger.block('Computing validation samples ...'):
+
+            self.validation_data = []
+            for datum in self.validation_set:
+                if not self.fom:
+                    mu, output = datum
```

Comment:
Now each element of the `validation_set` is already a pair containing a parameter and the output, right? That was not the case in the other reductor.